include/openthread/srp_client.h

```diff
@@ -561,6 +561,36 @@ otError otSrpClientSetDomainName(otInstance *aInstance, const char *aName);
  */
 const char *otSrpClientItemStateToString(otSrpClientItemState aItemState);
 
+/**
+ * This function enables/disables "service key record inclusion" mode.
+ *
+ * When enabled, SRP client will include KEY record in Service Description Instructions in the SRP update messages
+ * that it sends.
+ *
+ * This function is available when `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` configuration is enabled.
+ *
+ * @note KEY record is optional in Service Description Instruction (it is required and always included in the Host
+ * Description Instruction). The default behavior of SRP client is to not include it. This function is intended to
+ * override the default behavior for testing only.
+ *
+ * @param[in] aInstance  A pointer to the OpenThread instance.
+ * @param[in] aEnabled   TRUE to enable, FALSE to disable the "service key record inclusion" mode.
+ *
+ */
+void otSrpClientSetServiceKeyRecordInclusionEnabled(otInstance *aInstance, bool aEnabled);
```

Comment:
nit: this name is really long. Can we shorten like this?
```suggestion
void otSrpClientSetServiceKeyRecordEnabled(otInstance *aInstance, bool aEnabled);
```

Reply:
Renamed in last pushed commit.